### DISCUSSION TITLE
Get the value of the fields in the hidden section

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#7989] [Help]
How can I get the value of the fields in the hidden section? The section becomes hidden when the button is clicked, but its data should be received during the next operation.

**Comment by leandrocfe** [comment]
if the section is hidden, why would you like to get the values?

**Reply by ArtMin96** [reply]
It's like a wizard, but since I can't get what I need with a wizard, I try to get it with a simple form. I need to hide the first section and show the second one when the button is clicked. But when the form is submitted, it is necessary to get the data of the entire form.

**Reply by ArtMin96** [reply]
Section 1 (visible) **/** Section 2 (hidden)

Click "Next" button -> Hide section 1 **/** Show section 2
Click "Submit" -> get entire form state

**Reply by leandrocfe** [reply]
if you want to get all data, you can use `$this->form->getRawState()`. But rawState won't validate the inputs.
Not sure, but you could try `->extraAttributes(['class' => 'hidden'])` to control the frontend visibility.

**Reply by ArtMin96** [reply]
Thank you. In the case of a hidden class, extra space remains between elements.

![image](https://github.com/filamentphp/filament/assets/29732308/84481320-9c09-4d2e-9f70-f6e2b71f7d63)

**Reply by ArtMin96** [reply]
![7](https://github.com/filamentphp/filament/assets/29732308/1787254a-0a71-4cca-a0ad-586794971566)

**Reply by ArtMin96** [reply]
What if I use both? `getState` as a validator and `getRawState` to get the state?

**Reply by leandrocfe** [reply]
possible, I think. `$this->data` as well

**Comment by danharrin** [comment]
`hidden()->dehydrated()`?

**Reply by ArtMin96** [reply]
Thanks. Didn't work

**Reply by ArtMin96** [reply]
```php
 Section::make('section')
    ->hidden(fn (Component $livewire): bool => $livewire->propertyIsTrue)
    ->dehydrated()
    ->schema([]),
```

**Reply by leandrocfe** [reply]
what about a View layout? [view-components](https://filamentphp.com/docs/3.x/forms/layout/custom#view-components)

```php
use Filament\Forms\Components\View;

Step::make('step1')
    ->schema([
        View::make('custom-layout')
            ->schema([
                Section::make('section')
                    ->schema([
                        TextInput::make('field1'),
                    ])
            ]),
    ]),
Step::make('step2')
    ->schema([
        TextInput::make('field2'),
    ])
```
resources/views/custom-layout.blade.php
```html
<div class="{{ $this->propertyIsTrue ? 'block' : 'hidden' }}">
    {{ $getChildComponentContainer() }}
</div>
```

**Reply by ArtMin96** [reply]
Thank you for being so helpful. I will try

**Reply by gooner202** [reply]
Hey, do you have a solution for this?

**Reply by StaffCollab** [reply]
If `mutateFormDataBeforeCreate()` or these methods are not working for you. (it didn't work for me since I was in an Attach action, and I also have some other unique use cases) There is a simple trick to achieve the same results

Just use a `Hidden::make('name')->default('John')->dehydrated(true)` field. This will work, and if you optionally do need to customize the data, you can create a regular `TextInput::make()` field, and it will be synced with the hidden one automatically since both use the same `wire:model` behind the scene
